artifactory_cleanup/artifactorycleanup.py

```diff
@@ -65,12 +71,14 @@ def cleanup(self, block_ctx_mgr, test_ctx_mgr) -> Iterator[CleanupSummary]:
             try:
                 artifacts_size = sum([x["size"] for x in artifacts_to_remove])
                 print("Summary size: {}".format(artifacts_size))
-                yield CleanupSummary(
+                summary = CleanupSummary(
                     policy_name=policy.name,
                     artifacts_size=artifacts_size,
                     artifacts_removed=len(artifacts_to_remove),
                 )
-
+                if self.save_removed_artifacts_list:
+                    summary.removed_artifacts_list = artifacts_to_remove
+                yield summary
```

Comment:
LEt's always transfer that data, so Cleanup even doesn't have to worry about `self.save_removed_artifacts_list`
It's inmemory and cheep data